include/seqan3/index/fm_index_iterator.hpp

```diff
@@ -0,0 +1,521 @@
+
+        typename index_type::comp_char_type c = node.last_char + 1;
+        typename index_type::size_type _lb, _rb;
+
+        while (c < index->m_index.sigma &&
+               !sdsl::backward_search(index->m_index, parent_lb, parent_rb, index->m_index.comp2char[c], _lb, _rb))
+        {
+            ++c;
+        }
+
+        if (c != index->m_index.sigma)
+        {
+            node = {_lb, _rb, node.depth, c};
+            return true;
+        }
+        return false;
+    }
+
+    /*!\brief Returns an array of iterators pointing to the child nodes of the current iterator. Does not modify the
+     *        current iterator.
+     * \returns Array of iterators of size `alphabet_size(char_type)`, i.e. one iterator for each character. If the
+     *          current node does not have an edge for each character, the remaining positions in the array will be
+     *          filled with iterators pointing to the root.
+     *
+     * ### Complexity
+     *
+     * sigma * O(T_BACKWARD_SEARCH). The asymptotic running time is equal to enumerating all children using `down()` and
+     * `right()` but has a better cache performance. (TODO: to be verified)
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    auto children() const noexcept
+    {
+        assert(index != nullptr);
+
+        std::array<fm_index_iterator, alphabet_size_v<typename index_type::char_type>> result;
+
+        // TODO: this will be implemented much more efficiently in the future
+        uint8_t i = 0;
+        fm_index_iterator _it = *this;
+        if (_it.down())
+        {
+            do
+            {
+                result[i++] = _it;
+            } while (_it.right());
+        }
+
+        // fill the rest with iterators pointing to root
+        while (i < index_type::char_type::value_size)
+        {
+            result[i++] = fm_index_iterator(*this->index);
+        }
+
+        return result;
+    }
+
+    /*!\brief Returns the depth of the iterator node in the implicit suffix tree, i.e. the length of the sequence
+     *        searched.
+     * \returns Length of searched sequence.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    size_type depth() const noexcept
+    {
+        assert(index != nullptr);
+        assert(node.depth != 0 || (node.lb == 0 && node.rb == index->size() - 1)); // depth == 0 -> root node
+
+        return node.depth;
+    }
+
+    /*!\brief Checks whether the iterator is at the root node.
+     * \returns `true` if the iterator is pointing to the root node, `false` otherwise.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    bool is_root() const noexcept
+    {
+        assert(index != nullptr);
+
+        return depth() == 0;
+    }
+
+    // TODO: naming? maybe dereference-operator instead?
```

Comment:
I like the idea of using a dereference operator

Reply:
I'm not sure, for me it's clear what I expect from dereferencing the iterator, but a named method might be more verbose?

Reply:
Do you have a suggestion? I'm not really happy with `path_label()` :thinking:

Reply:
`get_path`, `path_representation` (which I find quite long) ?
I actually find `path_label` catchy